### PR TITLE
feat(playback config): handling of playback configuration and stream priority

### PR DESCRIPTION
**Pull request by dan-ziv**
- muted
- autoplay
- stream priority

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -130,51 +156,100 @@ class Player extends FakeEventTarget {
...
+   * Selects the engine to create based on a given configuration.
+   * @private
+   * @returns {boolean} - Whether a proper engine was found.
+   */
+  _selectEngine(): boolean {
+    let engineSelected = false;
+    if (this._config.sources) {
+      if (this._config.playback && this._config.playback.streamPriority) {
+        engineSelected = this._selectEngineByPriority();
+      } else {
```
Comment (yairans) [inline src/player.js]: we should have default priority

File: src/player.js [hunk 2]
```diff
@@ -130,53 +158,95 @@ class Player extends FakeEventTarget {
    * @static
    */
   static _defaultConfig(): Object {
-    return {};
+    return DefaultPlayerConfig;
```
Comment (OrenMe) [inline src/player.js]: Make this immutable, we don't want default config to change.
you can just serialize it via object.assign

File: src/player.js [hunk 3]
```diff
@@ -130,53 +158,95 @@ class Player extends FakeEventTarget {
...
   /**
-   * Select the engine to create based on the given configured sources.
-   * @param {Object} config - The configuration of the player instance.
+   * Selects the engine to create based on a given configuration.
+   * @private
+   * @returns {boolean} - Whether a proper engine was found.
+   */
+  _selectEngine(): boolean {
+    let engineSelected = false;
+    if (this._config.sources) {
```
Comment (OrenMe) [inline src/player.js]: We can always just do _selectEngineByPriority, cause we have the default setup.
Now just need to make sure the default config is deep cloned.

File: src/player.js [hunk 4]
```diff
@@ -560,4 +644,5 @@ class Player extends FakeEventTarget {
 // </editor-fold>
 }
 
-export default Player;
+const registerEngine = Player.registerEngine;
+export {registerEngine};
```
Comment (OrenMe) [inline src/player.js]: Let's remove for now, don't see a use case for external playback engine that we want to support without too much added complexity.

File: src/player.js (hunk 2 above)
Comment (dan-ziv) [inline src/player.js]: done

File: src/player.js (hunk 3 above)
Comment (dan-ziv) [inline src/player.js]: done

File: src/player.js (hunk 4 above)
Comment (dan-ziv) [inline src/player.js]: done

**Pull request merged**